botbot_plugins/plugins/jira.py

```diff
@@ -8,55 +9,79 @@
+    Returns the description of a JIRA Issue
 
         jira:{{projectname}}-{{issuenumber}}
     """
     config_class = Config
     
-    @listens_to_all(ur'(?:.*)\b(?P<project>\w+)-(?P<issue>\d+)\b(?:.*)')
-    def issue_lookup(self, line, project, issue):
-        """Lookup a specified jira issue
-
-            Usage:
-                Just mention the issue by its {{ISSUENAME}}
-                Eg:
-                    Can you please checkup on PROJECT-123
+    @listens_to_all(ur'(?:.*)\b(\w+-\d+)\b(?:.*)')
+    def issue_lookup(self, line):
         """
+        Lookup a specified JIRA issue
 
-        api_url = urljoin(self.config['jira_url'], self.config['rest_api_suffix'])
-        projects = json.loads(self.retrieve('projects'))
-        if project.upper() in projects and line.user != self.config['bot_name']:
-            issue_url = urljoin(api_url,"issue/{}-{}".format(project.upper(),(issue)))
-            response = requests.get(issue_url)
-            if response.status_code == 200:
-                response_text = json.loads(response.text)
-                name = response_text['key']
-                desc = response_text['fields']['summary']
-                return_url = urljoin(self.config['jira_url'],"projects/{}/issues/{}".format(project,name))
-                return "{}: {}\n{}".format(name,desc,return_url)
+        Usage:
+            Just mention the issue by its {{ISSUENAME}}
+            Eg:
+                Can you please checkup on PROJECT-123
+        """
+
+        if line.user not in (self.config['bot_name'],'github'):
+            api_url = urljoin(self.config['jira_url'], self.config['rest_api_suffix'])
+            projects = json.loads(self.retrieve('projects'))
+
+            queries = re.findall(r"\w+-\d+", line.text)
+            queries = [query.split("-") for query in queries]
+            reply = []
+
+            for query in queries:
+                if query[0] in projects:
+                    issue_url = urljoin(api_url,"issue/{}-{}".format(query[0],query[1]))
+                    response = requests.get(issue_url)
+
+                    if response.status_code == 200:
+                        response_text = json.loads(response.text)
+                        name = response_text['key']
+                        desc = response_text['fields']['summary']
+
+                        # Only post URL if issue isn't already mentioned as part of one
+                        if re.search(ur'(http)(\S*)/({})\b'.format(name), line.text):
+                            reply.append("{}: {}".format(name, desc))
+
+                        else:       
+                            return_url = urljoin(self.config['jira_url'], "browse/{}".format(name))
+                            reply.append("{}: {} {}".format(name, desc, return_url))
+                
+            # if Off topic then bot also replies with off topic                   
+            if re.search(r'^\[off\]', line.text):                     
```

Comment:
for our `off` messages, we accept any capitilization. You could use

    if line.text.lower().startswith("[off]")

instead